src/pages/Authentication/Signup.js

```diff
@@ -0,0 +1,195 @@
+import "./Authentication.css";
+import { useReducer } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { Box } from "../../components";
+import { setUserDetailsReducer } from "../../reducers";
+import { userSignup } from "../../utils/authenticationCalls";
+import { FaEye, FaEyeSlash } from "react-icons/fa";
+import { useAuth } from "../../contexts/AuthContext";
+
+export const Signup = () => {
+  const [userDetails, dispatchUserDetails] = useReducer(setUserDetailsReducer, {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    showPassword: false,
+  });
+
+  const {
+    firstName,
+    lastName,
+    email,
+    password,
+    confirmPassword,
+    showPassword,
+  } = userDetails;
+
+  const navigate = useNavigate();
+  const { dispatchAuth } = useAuth();
+  // const { dispatchData } = useData();
```

Comment:
remove commented out code if not required

Reply:
I have removed it in my next PR